BitwardenShared/Core/Autofill/Services/AutofillCredentialService.swift

```diff
@@ -203,10 +210,26 @@ class DefaultAutofillCredentialService {
             errorReporter.log(error: error)
         }
     }
+
+    /// Attempts to unlock the user's vault if it can be done without user interaction (e.g. if
+    /// the user uses never lock).
+    ///
+    /// - Parameter delegate: The delegate used for autofill credential operations.
+    ///
+    private func tryUnlockVaultWithoutUserInteraction(delegate: AutofillCredentialServiceDelegate) async throws {
+        let vaultTimeout = try await vaultTimeoutService.sessionTimeoutValue(userId: stateService.getActiveAccountId())
+        guard vaultTimeout == .never else { return }
+        try await delegate.unlockVaultWithNeverlockKey()
+    }
```

Comment:
🤔 Shouldn't we check if the vault is locked first in order to attempt to unlock it?

Reply:
Yeah, I'll add that.